src/graph/GraphViewServer.ts

```diff
@@ -150,13 +263,41 @@ export class GraphViewServer extends EventEmitter {
       }
     } catch (error) {
       // Shouldn't happen, just being defensive
+      console.error(error);
     }
 
     return message;
   }
 
   private async executeQuery(queryId: number, gremlinQuery: string): Promise<any[]> {
-    this.log(`Executing query #${queryId}: ${gremlinQuery}`);
+    const maxRetries = 3; // original try + this many extra tries
+    const retryDurationMs = 1000;
+    let iTry = 0;
+
+    while (true) {
+      iTry++;
+
+      try {
+        if (iTry > 1) {
+          this.log(`Retry #${iTry - 1} for query ${queryId}: ${truncateQuery(gremlinQuery)}`);
+        }
+        return await this._executeQueryCore(queryId, gremlinQuery);
+      } catch (err) {
+        if (this.isErrorRetryable(err)) {
+          if (iTry >= maxRetries) {
+            this.log(`Max retries reached for query ${queryId}: ${truncateQuery(gremlinQuery)}`);
+          } else {
+            continue;
```

Comment:
If I'm getting 'RequestRateTooLarge' errors - will trying again immediately work most of the time? Or do we need to wait a few seconds before trying again?

Reply:
Theoretically the time to wait is in the returned data, but it's embedded inside the error message. Not sure that I want to pick it out.  So far just the latency of trying again seems to be enough delay, but yeah I'll probably add some specific delay later.